### ISSUE TITLE
Further control over presentation of Problems

### ISSUE THREAD

**Issue by thomthom** [Microsoft/vscode#52128]
This relate to my feature request of https://github.com/Microsoft/vscode/issues/44141

The issue was resolved, but it only partially resolved my request. The use case I had was that I wanted to see squiggly underlines for the spellcheker in the editor view, but I didn't want to see the spelling issues reported in the Problems panel or Explorer view.

Currently one can control the type of diagnostic level for the spell checker extension;

```
"cSpell.diagnosticLevel": "Hint",
```

This do indeed hide silence the report in the Problems panel and Explorer view, but with the draw back of changing the squiggly underlines in the Editor to the ellipsis:

![image](https://user-images.githubusercontent.com/192418/41507209-47f4d9d2-722e-11e8-9634-b770f8573ee8.png)

Would it be possible to have some control for the user or extension developers to allow such control?

**Comment by thomthom** [comment]
Just realised that the Explorer panel already works as I want it: it doesn't list the warnings from the spellchecker. But the problem panel is filled with the spellchecker issues - which I don't find useful.

![image](https://user-images.githubusercontent.com/192418/41595400-f388f046-73c6-11e8-9039-4cbb2a49c9b9.png)

So maybe being able to configure a permanent filter for the Problems view could work? Either exclude per type (Hints, etc), or exclude per extension provider.

**Comment by alvinkiett84** [comment]
Get Outlook for Android<https://aka.ms/ghei36>

**Comment by jrieken** [comment]
@mjbvz Might be a candidate for `DiagnosticTag`

**Comment by mjbvz** [comment]
Yes, something like `DiagnosticTag.SuppressProblemsEntry` or `DiagnosticTag.OnlyShowInEditor` perhaps
